Neos.ContentRepository/Classes/Domain/Projection/Content/TraversableNodeInterface.php

```diff
@@ -0,0 +1,134 @@
+<?php
+namespace Neos\ContentRepository\Domain\Projection\Content;
+
+/*
+ * This file is part of the Neos.ContentRepository package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\ContentRepository\Domain\ValueObject\NodeName;
+use Neos\ContentRepository\Domain\ValueObject\NodePath;
+use Neos\ContentRepository\Domain\ValueObject\NodeTypeConstraints;
+use Neos\EventSourcedContentRepository\Domain\ValueObject\PropertyName;
+
+/**
+ * A convenience wrapper.
+ *
+ * Immutable. Read-only. With traversal operations.
+ *
+ * !! Reference resolving happens HERE!
+ */
+interface TraversableNodeInterface extends NodeInterface
+{
+    /**
+     * Retrieves and returns the parent node from the node's subgraph.
+     * Returns null if this is a root node.
+     *
+     * @return TraversableNodeInterface|null
+     */
+    public function findParentNode(): ?TraversableNodeInterface;
+
+    /**
+     * Retrieves and returns the node's path to its root node.
+     *
+     * @return NodePath
+     */
+    public function findNodePath(): NodePath;
```

Comment:
Because we say: "all operations which are slow should be named find*", vs. "all operations which operate on already-loaded data should be named get*". And fetching a node path is a potentially costly operation.

Reply:
Right, I remember! We should probably add that note to the `find*()` methods

Reply:
Absolutely :)

Reply:
adding note in next commit